### ISSUE TITLE
Feature Request: Nearby Bluetooth Devices

### ISSUE THREAD

**Issue by pjv** [dustinrue/ControlPlane#389]
It would be great if the Bluetooth evidence source included nearby (i.e. active but not connected) bluetooth devices like the "Nearby WiFi Network" evidence source does, and then allow rules to be triggered based on whether a particular device is "nearby".

Use case: I want to create a rule+action that will automatically connect to a bluetooth headset whenever it is turned on in the vicinity of my mac (can't believe Apple doesn't provide this capability within the Bluetooth preference pane).

See also #326

**Comment by dmland** [comment]
Is it appropriate to +1 this request?

I recall that MarcoPolo supported bluetooth detection that did not require connecting (which is so stupidly complex with a Mac and iPhone that I might as well just manually do the things I want ControlPlane to do for me), but merely the presence of a specific Bluetooth device. This was so long ago that the device in question was a Blackberry!

**Comment by packwidth** [comment]
I agree that this would be nice. I'm interested in connecting to a Bluetooth PAN (my phone) when there's otherwise no network access available (so far so good)... and reestablishing the connection if I move too far away and then return.

This last part is difficult without setting CP to deliberately bounce between contexts (so as to connect to PAN on both departure and arrival).

**Comment by SeAlgoAsoma** [comment]
Chiming in a little later to say still would be nice.

**Comment by dustinrue** [comment]
Closing as not planned. Continuously scanning for nearby unpaired Bluetooth devices requires active radio inquiry (IOBluetoothDeviceInquiry) which is disruptive to the 2.4GHz spectrum and battery life. For the common use case of reacting to a known device — such as a headset powering on — the existing BluetoothSensor already handles this: paired devices emit a boolean reading that transitions true/false as they connect and disconnect.
